### ISSUE TITLE
Ankkurilinkit toiseen saman dokumentin otsikkoon eivät tottele määritettyjä ankkureita

### ISSUE THREAD

**Issue by saviit** [TIM-JYU/TIM#3325]
Ankkurilinkit toiseen saman dokumentin otsikkoon eivät aina tottele määritettyjä ankkureita: esimerkki <https://tim.jyu.fi/view/users/viitanen-sami/testikansio/anchor/anchortest>

Voi toistaa seuraavilla askeleilla:
1. dokumenttiin lisätään vähintään kaksi saman nimistä otsikkoa
2. ensimmäiselle otsikolle **a** määritetään otsikon nimestä poikkeava id syntaksilla `## Otsikko {#eka-otsikko}`
3. toiselle otsikolle **b** määritetään otsikon nimeä mukaileva id syntaksilla `## Otsikko {#otsikko}` siten, että otsikon nimessä olevat erikoismerkit poistetaan ja välilyönnit korvataan väliviivoilla
4. dokumenttiin lisätään linkki toisen otsikon **b** id:llä syntaksilla `(linkki)[#otsikko]`
5. nyt juuri lisätty linkki viekin tuonne otsikkoon **a**, vaikka sen pitäisi viedä otsikkoon **b**.

ts. erikseen asetettuja id:tä ei kunnioiteta.

Hiukan samaan viittaavaa ongelmaa on kortissa #2102. Tässä kortissa kuvattu taitaa olla kuitenkin vielä oma tapauksensa.

**Comment by vesal** [comment]
>  5. nyt juuri lisätty linkki viekin tuonne otsikkoon a, vaikka sen pitäisi viedä otsikkoon b.

Joo, tuo tulee koska jokainen otsikko saa automaattisesti id:n joka on sen
otsikon teksti.  Eli nyt ei ole muuta mahdollisuutta kun antaa itse tehdyille
id:lle tekstej joita ei esiinny missään otsikossa.  Tuo vaatisi aika
perusteellista muutosta että automaattisesti tehtävät linkit poistettaisiin
ja se rikkoisi monia dokumentteja.

**Comment by saviit** [comment]
Ok, joo. Tuo pitää varmaan dokumentoida jotenkin selkokielisesti tuonne TIMin ohjeisiin, muuten tuo tulee ikuisesti aiheuttamaan ihmetystä.

**Comment by vesal** [comment]
> Se ongelmahan tässä ei ollut sisällysluettelo, vaan se että nuo ankkurilinkit itse dokumentissa ei toimi odotetulla tavalla (eli että itse asetettu ankkuri olisi aina ensisijainen).

Eikös se ole periaattessa HTM:ssä kielletty että on kaksi samannimistä id:tä.
Mutta pointtini olis se, että jos otikolle on annettu itse id, niin käyttääkö
sislu sitä vaiko tekstistä generoitua.  Oikein olisi käyttää siinä
itse tehtyä, niin osoitteen voi kopioida sislustakin sellaisiin.

Se tietysti voisi olla pieni korjatus johonkin että jos jollekinon itse annettu
id, niin automaattsietsi ei generoida.  Mutta tosiaan se voisi
rikkoa ison kasan dokumentteja, jossa on linkkejä laitettu
autom generoituun osoitteeseen ja sitten joku lisää oman id:n.

**Comment by saviit** [comment]
Sisällysluettelo ei mielestäni käytä itse tehtyjä id:tä (en tarkistanut koodista, mutta tämän näkee selaimessakin tuolla testisivulla <https://tim.jyu.fi/view/users/viitanen-sami/testikansio/anchor/anchortest>). Testisivulla on kolme samaa otsikkoa: ekalla ja kolmannella otsikolla itse annetut id:t, ja tokassa otsikossa on automaattinen. Sislu tekee silti nuo kaavalla otsikon nimi + järjestysnumero.

Jos olen ymmärtänyt oikein niin tuon sisällysluettelon muuttaminen ei kuitenkaan korjaa sitä, että dokumentissa olevaa ankkurilinkkiä klikatessa mennään väärän otsikon kohdalle? Ellei kyseessä ole esim jotain sellaista, että sisällysluetteloa generoidessa munklaillaan id:t ristiin? Pitää varmaan seuraavaksi tutkia koodin puolelta että missä siellä mennään mönkään tuon kanssa.
